### PR TITLE
Finalize 2nd order schedulers

### PR DESCRIPTION
**Pull request by patrickvonplaten**
Corrects Heun scheduler (previously one unnecessary step was done).
Also adds two additional 2nd order schedulers:
- https://github.com/crowsonkb/k-diffusion/blob/5b3af030dd83e0297272d861c19477735d0317ec/k_diffusion/sampling.py#L188
- https://github.com/crowsonkb/k-diffusion/blob/5b3af030dd83e0297272d861c19477735d0317ec/k_diffusion/sampling.py#L219

Ran all schedulers for:
- v-prediction (SD 2 768): https://huggingface.co/datasets/patrickvonplaten/images/tree/main/a_astronaut_riding_horse
- epsilon-prediction (SD 2 base): https://huggingface.co/datasets/patrickvonplaten/images/tree/main/a_astronaut_riding_horse_v2_base_sd

Results match k-diffusion implementation.

### CODE REVIEW

File: src/diffusers/schedulers/scheduling_heun.py [hunk 1]
```diff
@@ -212,6 +214,7 @@ def step(
             self.sample = None
 
         prev_sample = sample + derivative * dt
+        print("x_2", prev_sample.abs().sum())
```
Comment (patrickvonplaten) [inline src/diffusers/schedulers/scheduling_heun.py:217]:
```suggestion
```

File: src/diffusers/schedulers/scheduling_heun.py [hunk 2]
```diff
@@ -131,7 +133,7 @@ def set_timesteps(
         self.init_noise_sigma = self.sigmas.max()
 
         timesteps = torch.from_numpy(timesteps)
-        timesteps = torch.cat([timesteps[:1], timesteps[1:].repeat_interleave(2), timesteps[-1:]])
```
Comment (patrickvonplaten) [inline src/diffusers/schedulers/scheduling_heun.py:134]: This was incorrect previously, but didn't change the output of the image. Can be removed so since it's an unnecessary extra step
